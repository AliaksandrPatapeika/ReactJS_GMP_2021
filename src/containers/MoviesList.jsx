import '../less/containers/MoviesList.less';

import React from 'react';

import {MockMovies} from '../mocks/mockData';
import MovieCard from './MovieCard';

const MoviesList = () => (
  <div className="moviesListContainer">
    {MockMovies.map((item) => (
      <MovieCard
        movie={item}
        key={item.id}
      />
    ))}
  </div>
);

export default MoviesList;
