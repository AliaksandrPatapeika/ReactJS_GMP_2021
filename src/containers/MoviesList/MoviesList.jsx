import './MoviesList.less';

import React from 'react';

import {MockMovies} from '../../mocks/mockData';
import MovieCard from '../MovieCard';

const MoviesList = () => {
  const movies = MockMovies;

  return (
    <div className="moviesListContainer">
      {movies.map((item) => (
        <MovieCard
          movie={item}
          key={item.id}
        />
      ))}
    </div>
  );
};

export default MoviesList;
