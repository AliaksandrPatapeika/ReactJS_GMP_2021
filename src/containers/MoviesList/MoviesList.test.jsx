import React from 'react';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import MockMovies from '../../tests/mocks/mockData';
import MoviesList from './MoviesList';

describe('MoviesList', () => {
  test('renders correctly', () => {
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        movies: MockMovies
      }
    };
    const store = mockStore(initialState);
    const moviesList = renderer
      .create(
        <Provider store={store}>
          <Router>
            <MoviesList />
          </Router>
        </Provider>
      )
      .toJSON();

    expect(moviesList).toMatchSnapshot();
  });
});
