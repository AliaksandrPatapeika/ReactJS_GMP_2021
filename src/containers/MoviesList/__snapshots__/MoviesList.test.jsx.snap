// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList renders correctly 1`] = `
<div
  className="moviesListContainer"
>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="imageButton"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="item"
        className="imageItem"
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie button"
        src="test-file-stub"
      />
    </button>
    <div
      className="descriptionContainer"
    >
      <div
        className="itemTitle"
      >
        Pulp Fiction
      </div>
      <div
        className="itemReleaseDate"
      >
        2004
      </div>
    </div>
    <div
      className="itemGenre"
    >
      Action & Adventure
    </div>
  </div>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="imageButton"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="item"
        className="imageItem"
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie button"
        src="test-file-stub"
      />
    </button>
    <div
      className="descriptionContainer"
    >
      <div
        className="itemTitle"
      >
        Bohemian Rhapsody
      </div>
      <div
        className="itemReleaseDate"
      >
        2003
      </div>
    </div>
    <div
      className="itemGenre"
    >
      Drama, Biography, Music
    </div>
  </div>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="imageButton"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="item"
        className="imageItem"
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie button"
        src="test-file-stub"
      />
    </button>
    <div
      className="descriptionContainer"
    >
      <div
        className="itemTitle"
      >
        Kill Bill: Vol 2
      </div>
      <div
        className="itemReleaseDate"
      >
        1994
      </div>
    </div>
    <div
      className="itemGenre"
    >
      Oscar winning Movie
    </div>
  </div>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="imageButton"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="item"
        className="imageItem"
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie button"
        src="test-file-stub"
      />
    </button>
    <div
      className="descriptionContainer"
    >
      <div
        className="itemTitle"
      >
        Avengers: War of Infinity
      </div>
      <div
        className="itemReleaseDate"
      >
        2004
      </div>
    </div>
    <div
      className="itemGenre"
    >
      Action & Adventure
    </div>
  </div>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="imageButton"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="item"
        className="imageItem"
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie button"
        src="test-file-stub"
      />
    </button>
    <div
      className="descriptionContainer"
    >
      <div
        className="itemTitle"
      >
        Inception
      </div>
      <div
        className="itemReleaseDate"
      >
        2003
      </div>
    </div>
    <div
      className="itemGenre"
    >
      Action & Adventure
    </div>
  </div>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="imageButton"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="item"
        className="imageItem"
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled=""
      id=""
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie button"
        src="test-file-stub"
      />
    </button>
    <div
      className="descriptionContainer"
    >
      <div
        className="itemTitle"
      >
        Reservoir dogs
      </div>
      <div
        className="itemReleaseDate"
      >
        1994
      </div>
    </div>
    <div
      className="itemGenre"
    >
      Oscar winning Movie
    </div>
  </div>
</div>
`;
